lib/srv/monitor.go

```diff
@@ -71,6 +71,8 @@ type ConnectionMonitorConfig struct {
 	ServerID string
 	// Emitter allows events to be emitted.
 	Emitter apievents.Emitter
+	// EmitterContext is long-lived context suitable to be used with Emitter
+	EmitterContext context.Context
```

Comment:
Obligatory https://github.com/golang/go/wiki/CodeReviewComments#contexts. (Although, tbh, I'm almost giving up quoting this one.)

Reply:
We kind of violate this nearly everywhere, though. For audit events, in particular, I think we should always use server exit context or equivalent. Unless the server is shutting down, hard, we should still try to emit the event. 

I'd argue this kind of sharing is fine, too, at least for the purpose of synchronising the lifetimes of different services. Sharing credentials would be much more icky.

Reply:
Yep.

Reply:
Still, thanks for the reminder, I have actually forgotten about this style issue.